lightkurve/targetpixelfile.py

```diff
@@ -816,6 +818,70 @@ def to_fits(self, output_fn=None, overwrite=False):
             output_fn = "{}-targ.fits".format(self.keplerid)
         self.hdu.writeto(output_fn, overwrite=overwrite, checksum=True)
 
+    def get_hdu(image, extension):
+        """Checks if an image is a FITS single image, or a FITS file,
+        and returns the appropriate PrimaryHDU.
+
+        Attributes
+        ----------
+        image : fits.ImageHDU or fits.HDUList object
+            FITS filename path or ImageHDU object to get the data from.
+        extension: integer 0, or 1
+            Determines which file extension the data is in, depending on the file
+            type and source
+        Returns
+        -------
+        hdu : astropy.PrimaryHDU
+            PrimaryHDU of the image inputted
+        """
+        if isinstance(image, fits.ImageHDU):
+            hdu = image
+        elif isinstance(image, fits.HDUList):
+            hdu = image[extension]
+        else:
+            hdu = fits.open(image)[extension]
+        return hdu
+
+    def get_WCS(mid_img,hdu,position,extension): #name of this?? add doc string
+        """Calculates the position correction values (both in row and column) and
+        WCS for an image relative to the reference image, defined as the middle
+        cadence of the FITS file.
+
+        Attributes
+        ----------
+        hdu : astropy.PrimaryHDU
+            PrimaryHDU of the image being looked at
+        mid_img : fits.ImageHDU
+            middle layer of the FITS file used as a reference image
+        Returns
+        -------
+        poscorr1 :  float32
+            Position correction in the vertical direction (column)
+        poscorr2 :  float32
+            Position correction in the horizontal direction (row)
+        wcs1 :  astropy.wcs.WCS
+            WCS object for this specific hdu
+        """
+        # Calculate reference WCS coords from the middle image to use later to
+        # do pos_corr calcs. Get the middle pixel column and row.
+        mid_hdu = KeplerTargetPixelFile.get_hdu(mid_img, extension)
+        wcs = WCS(mid_hdu)
+        pixelpos = skycoord_to_pixel(position, wcs=wcs)
+        column = int(np.round(pixelpos[0]))
+        row = int(np.round(pixelpos[1]))
```

Comment:
You can probably reduce this to one line, something like:
```
col, row = int(skycoord_to_pixel(position, wcs=wcs))
```

Reply:
Actually, we need the round in there, because otherwise it just takes the integer value, which means, e.g., 5.6 --> 5 instead of 6